util/logtools/anonymizer.ts

```diff
@@ -242,6 +264,70 @@ export default class Anonymizer {
     return splitLine.join('|');
   }
 
+  // This method assumes that repeating fields use the current `CombatantMemory` structure
+  // of key/value pairs, e.g. Key1|Value1|...|KeyN|ValueN|.
+  // If other structures are used, this method will need to be updated.
+  private anonymizeRepeatingFields(
+    rfToAnonymize: { [key: string | number]: string | number | null },
+    splitLine: string[],
+    startIdx: number,
+    notifier: Notifier,
+  ): string[] {
+    for (const [idFieldKey, nameFieldKey] of Object.entries(rfToAnonymize)) {
+      let idFieldKeyIdx: number;
+      let idIsRepeatingField = true; // default assumption
+
+      if (typeof idFieldKey === 'number' || !isNaN(parseInt(idFieldKey))) {
+        idIsRepeatingField = false;
+        idFieldKeyIdx = parseInt(idFieldKey);
+      } else
+        idFieldKeyIdx = splitLine.findIndex((field, idx) =>
+          idx >= startIdx && field === idFieldKey
+        );
+
+      // ID field is not present; not an error condition for repeating fields.
+      if (idFieldKeyIdx === -1)
+        continue;
+
+      const idFieldValueIdx = idIsRepeatingField ? idFieldKeyIdx + 1 : idFieldKeyIdx;
+      const idFieldValue = (splitLine[idFieldValueIdx] ?? '').toUpperCase();
+
+      // If not a playerId value, ignore.
+      // TODO: Could add warnings here for malformed values, but there are many cases
+      // where it's fine if it's not a playerId, so ignoring seems safer/less noisy.
+      if (idFieldValue.length !== 8 || !idFieldValue.startsWith('1'))
+        continue;
+
+      const fakePlayerId = this.anonMap[idFieldValue] ??= this.addNewPlayer();
+      splitLine[idFieldValueIdx] = fakePlayerId;
+
+      // If no name field is paired with this id field, we're done.
+      if (nameFieldKey === null)
+        continue;
+
+      const nameFieldKeyIdx = splitLine.findIndex((field, idx) =>
+        idx >= startIdx && field === nameFieldKey
+      );
+      if (nameFieldKeyIdx === -1)
+        continue;
+
+      const nameFieldValueIdx = nameFieldKeyIdx + 1;
+      const playerName = splitLine[nameFieldValueIdx];
+      if (playerName === undefined || playerName === '') {
+        notifier.warn(`expected player name after '${nameFieldKey}'`, splitLine);
+        continue;
+      }
+      const fakePlayerName = this.playerMap[fakePlayerId];
+      if (fakePlayerName === undefined) {
+        splitLine[nameFieldValueIdx] = '';
+        continue;
+      }
+      splitLine[nameFieldValueIdx] = fakePlayerName;
```

Comment:
There's a lot of extra code here for something that could be done directly...

```suggestion
      splitLine[nameFieldValueIdx] = this.playerMap[fakePlayerId] ?? '';
```

Reply:
> There's a lot of extra code here for something that could be done directly...

I originally had more code after that block that necessitated an explicit `continue`, but I cleaned it up and forgot to simplify this.